internal/queue/mq.go

```diff
@@ -43,69 +43,54 @@ func (mq *MessageQueue) Write(msg interface{}) error {
 	return nil
 }
 
-// Read will read a value from the queue if available, otherwise return an error.
-func (mq *MessageQueue) Read() (interface{}, error) {
+// Dequeue will read a value from the queue if available, else it will wait for a
+// value to become available. This will block forever if nothing gets written or until
+// the queue gets closed.
```

Comment:
The last reads kinda weird
maybe?
> If nothing is written, this will block until the queue is closed

Reply:
Swapped the wording, let me know if this reads better!